flocx_ui/static/dashboard/project/flocx/contract-list/contract-list.controller.js

```diff
@@ -0,0 +1,61 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('horizon.dashboard.project.flocx')
+    .controller('ContractListController', ContractListController);
+
+  ContractListController.$inject = [
+    'horizon.app.core.openstack-service-api.flocx'
+  ];
+
+  function ContractListController(flocx) {
+    var ctrl = this;
+
+    ctrl.allContracts = [];
+    ctrl.contracts = [];
+    ctrl.pane = 'active';
+
+    ctrl.showPane = showPane;
+
+    // Event Handlers
+
+    function showPane (pane) {
+      ctrl.pane = pane;
+      filterContracts(ctrl.allContracts); // re-filter contracts
+    }
+
+    // Initialize Controllers
+
+    init();
+
+    function init () {
+      retrieveContracts();
+    }
+
+    function retrieveContracts () {
+      // Get the list of contracts from the flocx-market
+      flocx.getContracts().then(onGetContracts);
+    }
+
+    function onGetContracts (contracts) {
+      ctrl.allContracts = contracts;
+      filterContracts(contracts); // Filter the contracts depending on the active tab
+    }
+
+    function filterContracts (contracts) {
+      ctrl.contracts = contracts.filter(filter).map(function (contract) {
+        contract.start_time = new Date(contract.start_time).toLocaleString();
+        contract.end_time = new Date(contract.end_time).toLocaleString();
+        return contract;
+      });
+    }
+
+    function filter (contract) {
+      if (ctrl.pane === 'expired') {
```

Comment:
Use a global variable called 'expired' and use that for conditional statements. There are some talks that are going on in terms of identifying status. So, in case the status name is changed in the future, we will have to change only one statement. The ideal way would be to declare all the variable in a file and import that file whenever it is required.